src/features/game/quiz/GameQuizManager.ts

```diff
@@ -0,0 +1,214 @@
+    const header = new Phaser.GameObjects.Text(
+      scene,
+      screenSize.x - QuizConstants.textPad,
+      QuizConstants.y,
+      questionPrompt,
+      quizTextStyle
+    ).setOrigin(1.0, 0.0);
+
+    const quizHeaderBg = new Phaser.GameObjects.Rectangle(
+      scene,
+      screenSize.x,
+      QuizConstants.y - QuizConstants.textPad,
+      QuizConstants.width * quizPartitions,
+      header.getBounds().bottom * 0.5 + QuizConstants.textPad,
+      HexColor.darkBlue,
+      0.8
+    ).setOrigin(1.0, 0.0);
+
+    const quizBg = new Phaser.GameObjects.Rectangle(
+      scene,
+      screenSize.x,
+      QuizConstants.y - QuizConstants.textPad,
+      QuizConstants.width * quizPartitions,
+      quizHeaderBg.getBounds().bottom * 0.5 + (quizHeight + 0.5) * QuizConstants.yInterval,
+      HexColor.lightBlue,
+      0.2
+    ).setOrigin(1.0, 0.0);
+
+    quizContainer.add([
+      quizBg,
+      quizHeaderBg,
+      header,
+      quizQuestionBox,
+      quizQuestionWriter.container
+    ]);
+
+    const buttonPositions = calcListFormatPos({
+      numOfItems: choices.length,
+      xSpacing: 0,
+      ySpacing: QuizConstants.yInterval
+    });
+
+    GameGlobalAPI.getInstance().addToLayer(Layer.Dialogue, quizContainer);
+
+    //Create options for users to select
+    const activateQuizContainer: Promise<any> = new Promise(resolve => {
+      quizContainer.add(
+        choices.map((response, index) =>
+          createButton(scene, {
+            assetKey: ImageAssets.mediumButton.key,
+            message: response.text,
+            textConfig: QuizConstants.textConfig,
+            bitMapTextStyle: quizOptStyle,
+            onUp: async () => {
+              quizContainer.destroy();
+              const isCorrect = index === question.answer ? 1 : 0;
+              if (response.reaction) {
+                await this.showReaction(response.reaction);
+              }
+              resolve(isCorrect);
+            }
+          }).setPosition(
+            screenSize.x -
+              QuizConstants.width / 2 -
+              QuizConstants.width * (quizPartitions - Math.floor(index / 5) - 1),
+            (buttonPositions[index][1] % (5 * QuizConstants.yInterval)) +
+              quizHeaderBg.getBounds().bottom +
+              75
+          )
+        )
+      );
+    });
+
+    const response = await activateQuizContainer;
+
+    // Animate in
+    quizContainer.setPosition(screenSize.x, 0);
+    SourceAcademyGame.getInstance().getSoundManager().playSound(SoundAssets.notifEnter.key);
+    scene.add.tween({
+      targets: quizContainer,
+      alpha: 1,
+      ...rightSideEntryTweenProps
+    });
+    await sleep(rightSideEntryTweenProps.duration);
+
+    // Animate out
+    SourceAcademyGame.getInstance().getSoundManager().playSound(SoundAssets.notifExit.key);
+    scene.add.tween({
+      targets: quizContainer,
+      alpha: 1,
+      ...rightSideExitTweenProps
+    });
+
+    //await sleep(rightSideExitTweenProps.duration);
```

Comment:
Should this be in?

Reply:
Comment addressed: commit 1feee8bb7c87dabf2c4032f1cd3d91e5e9065d8c